amulet/deployer.py

```diff
@@ -429,7 +428,8 @@ def setup(self, timeout=600, cleanup=True):
             with self.deploy_w_timeout(timeout):
                 subprocess.check_call(shlex.split(cmd))
 
-        self.sentry = Talisman(self.services, timeout=timeout)
+        self.sentry = Talisman(self.services)
+        self.sentry.wait_for_status(self.sentry.juju_env, self.services, timeout)
```

Comment:
I was thinking that we'd call self.wait_for_status() from within sentry.wait(), and thereby eliminate all implicit waiting from the setup() call. Any reason not to do that?